code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -53,6 +54,21 @@
 		else if(prob(5))
 			emote("snuffles")
 
+
+	if(!isUnconscious())
+		var/list/can_see() = view(src, 5) //Decent radius, not too large so they're attracted across rooms, but large enough to attract them to mousetraps
+
+		if(!food_target)
+			for(var/obj/item/weapon/reagent_containers/food/snacks/C in can_see)
+				food_target = C
+				break
+		if(!(food_target in can_see))
+			food_target = null
+		if(food_target)
+			if(Adjacent(food_target))
+				food_target.attack_animal(src)
+			step_towards(src, food_target)
```

Comment:
Fair point, I'll have the step_towards before the adjacent check next chance I get